src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.GCDesc.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    partial class EETypeNode
+    {
+        private int GCDescSize
+        {
```

Comment:
It will be called two or three times. I was thinking about caching too, but we will likely see it in the profile if it's a problem.

> The encoded GCDesc would also work nicely as a key for GCStaticEETypeNode

I plan to use GCPointerMap for that - that's why it implements IEquatable. That part will be fixed as part of #1213. I'm trying to avoid large refactorings right now and coordinate with @nattress on that.